test/acvp_mlkem.c

```diff
@@ -73,23 +79,22 @@ static int decode_hex(const char *prefix, unsigned char *out, size_t out_len,
   size_t i;
   size_t hex_len = strlen(hex);
   size_t prefix_len = strlen(prefix);
-
   /*
    * Check that hex starts with `prefix=`
    * Use memcmp, not strcmp
    */
   if (hex_len < prefix_len + 1 || memcmp(prefix, hex, prefix_len) != 0 ||
       hex[prefix_len] != '=')
   {
-    goto hex_usage;
```

Comment:
Why was this removed?

Reply:
NIST added tests with wrong key lengths, so we actually need to handle it when decoding fails. For consistency it was changed everywhere.

Reply:
It's not consistent at the moment? There is still `goto hex_usage` below. Let's keep the 1st and 3rd occasion, and only change the second which is about length encoding.